pyathena/calc_cursor.py

```diff
@@ -0,0 +1,200 @@
+    @query_id.setter
+    def query_id(self, val: str) -> None:
+        self._calculation_id = val
+
+    @staticmethod
+    def get_default_engine_configuration() -> Dict[str, Any]:
+        return {
+            "CoordinatorDpuSize": 1,
+            "MaxConcurrentDpus": 2,
+            "DefaultExecutorDpuSize": 1,
+        }
+
+    def _get_session_status(self, session_id: str):
+        request: Dict[str, Any] = {"SessionId": session_id}
+        try:
+            response = retry_api_call(
+                self._connection.client.get_session_status,
+                config=self._retry_config,
+                logger=_logger,
+                **request,
+            )
+        except Exception as e:
+            _logger.exception("Failed to get session status.")
+            raise OperationalError(*e.args) from e
+        else:
+            return AthenaSession(response)
+
+    def _start_session(self) -> str:
+        request: Dict[str, Any] = {
+            "WorkGroup": self._work_group,
+            "EngineConfiguration": self._engine_configuration,
+        }
+        if self._session_description:
+            request.update({"Description": self._session_description})
+        if self._notebook_version:
+            request.update({"NotebookVersion": self._notebook_version})
+        if self._session_idle_timeout_minutes:
+            request.update({"SessionIdleTimeoutInMinutes": self._session_idle_timeout_minutes})
+        try:
+            session_id: str = retry_api_call(
+                self._connection.client.start_session,
+                config=self._retry_config,
+                logger=_logger,
+                **request,
+            )["SessionId"]
+        except Exception as e:
+            _logger.exception("Failed to start session.")
+            raise OperationalError(*e.args) from e
+        else:
+            while True:
+                session_status = self._get_session_status(session_id)
+                if session_status.state in [AthenaSession.STATE_IDLE]:
+                    return session_id
+                elif session_status in [
+                    AthenaSession.STATE_TERMINATED,
+                    AthenaSession.STATE_DEGRADED,
+                    AthenaSession.STATE_FAILED,
+                ]:
+                    raise OperationalError(session_status.state_change_reason)
+                else:
+                    time.sleep(self._poll_interval)
+
+    def _terminate_session(self) -> None:
+        request = {"SessionId": self._session_id}
+        try:
+            retry_api_call(
+                self._connection.client.terminate_session,
+                config=self._retry_config,
+                logger=_logger,
+                **request,
+            )
+        except Exception as e:
+            _logger.exception("Failed to terminate session.")
+            raise OperationalError(*e.args) from e
+
+    def __poll(self, query_id: str) -> Union[AthenaQueryExecution, AthenaCalculationExecution]:
```

Comment:
Would this actually return `AthenaQueryExecution`?

Reply:
SparkCursor only returns AthenaCalculationExecution, but the parent class BaseCursor may return AthenaQueryExecution, so I specify two classes in Union.